c2rust-transpile/src/translator/operators.rs

```diff
@@ -950,13 +950,58 @@ impl<'c> Translation<'c> {
                                 TranslationError::generic("Address-of should return a pointer")
                             })?;
 
+                    let expr_kind = &self.ast_context.index(arg).kind;
+                    let translate_as_macro = self
+                        .convert_const_macro_expansion(ctx, arg, None)
+                        .ok()
+                        .flatten()
+                        .is_some();
+
+                    if let (
+                        &CExprKind::Literal(
+                            literal_cqual_type,
+                            CLiteral::String(ref bytes, element_size @ 1),
+                        ),
+                        true,
+                        false,
+                    ) = (
```

Comment:
I think a comment is warranted here to explain exactly why we translate this case differently; something like ``// Because we translate string literals into temporary `c_char` arrays, we should not take the address of such a translation in a `const` context. Instead, cast the string literal directly.``